remove tuple_parser

nairbv opened the pull request:
Summary:

I could only find one usage of TupleParser and it doesn't seem worth maintaining just for that one usage.


Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30659 remove tuple_parser**

Differential Revision: [D18795979](https://our.internmc.facebook.com/intern/diff/D18795979)

nairbv committed:
remove tuple_parser

[ghstack-poisoned]

bddppq commented on torch/csrc/autograd/functions/init.cpp:
```diff
@@ -10,20 +10,21 @@
-    parser.parse(msg, "msg");
-    parser.parse(num_inputs, "num_inputs");
 
+    TORCH_CHECK(PyTuple_GET_SIZE(args) >= 2, "Too few arguments");
```
I think here should check `== 2`?

nairbv replied:
thanks!

nairbv committed:
Update on "remove tuple_parser"

Summary:

I could only find one usage of TupleParser and it doesn't seem worth maintaining just for that one usage.






[ghstack-poisoned]

nairbv committed:
Update on "remove tuple_parser"

Summary:

I could only find one usage of TupleParser and it doesn't seem worth maintaining just for that one usage.






[ghstack-poisoned]

bddppq commented on torch/csrc/autograd/functions/init.cpp:
```diff
@@ -10,20 +10,21 @@
-    parser.parse(msg, "msg");
-    parser.parse(num_inputs, "num_inputs");
 
+    TORCH_CHECK(PyTuple_GET_SIZE(args) == 2, "Requires two arguments");
```
nit: also put the actual size in the error message?

bddppq reviewed (approved):
LGTM

nairbv committed:
Update on "remove tuple_parser"

Summary:

I could only find one usage of TupleParser and it doesn't seem worth maintaining just for that one usage.






[ghstack-poisoned]

nairbv committed:
Update on "remove tuple_parser"

Summary:

I could only find one usage of TupleParser and it doesn't seem worth maintaining just for that one usage.






[ghstack-poisoned]
